aws-android-sdk-core/src/test/java/com/amazonaws/util/json/JsonUtilsTest.java

```diff
@@ -58,6 +76,28 @@ public void testJsonToMap() {
         assertNull("object is ignored", map.get("object"));
     }
 
+    @Test
+    public void testJsonToMapWithList() {
+        Map<String, String> map = JsonUtils.jsonToStringMapWithList(new StringReader(JSON_STRING_WITH_ARRAYS));
+        assertEquals("string value", "string", map.get("string"));
```

Comment:
Instead of using all of the individual `assertFoo(...)`s, I recommend using a strategy like:
1. Prepare a `Map` as input.
2. Prepare a JSON string, as below, that you can assert against.
3. Make a single assert call.

```
assertEquals(
    new JSONObject()
        .put("logn value", map.get("..."))
        // etc
        .toString(),
    map
);
```

Reply:
I do not really want to take a direct dependency on any Json library here, and we do not have wrappers around JsonObject provided by any libraries. I do not think this warrants an addition of that size at this point, it matches the styling of the code around it.

Reply:
`org.json.JSONObject` is [baked into Android](https://developer.android.com/reference/org/json/JSONObject). [`JsonObject`](https://www.javadoc.io/doc/com.google.code.gson/gson/2.8.5/com/google/gson/JsonObject.html) is part of Gson.